File: WinForm/AutomaticDrawingLib/AI/Vertex.vb
Imports System.Drawing
Imports System.Numerics
''' <summary>
''' 具有位置、颜色和大小的顶点
''' </summary>
Public Class Vertex
    ''' <summary>
    ''' 位置
    ''' </summary>
    Public Property Position As Vector2
    ''' <summary>
    ''' 颜色
    ''' </summary>
    Public Property Color As Color
    ''' <summary>
    ''' 大小
    ''' </summary>
    Public Property Size As Single
    ''' <summary>
    ''' 位置X分量
    ''' </summary>
    Public ReadOnly Property X As Integer
        Get
            Return Position.X
        End Get
    End Property
    ''' <summary>
    ''' 位置Y分量
    ''' </summary>
    Public ReadOnly Property Y As Integer
        Get
            Return Position.Y
        End Get
    End Property
End Class

File: AutoPaint.Core/Fundamental/Vertex/Vertex.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Numerics;
using System.Text;

namespace AutoPaint.Core
{
    public class Vertex
    {
        public Vector2 Position { get; set; }
        public Color Color { get; set; }
        public float Size { get; set; }
        public int LayerIndex { get; set; }

        public float X => Position.X;
        public float Y => Position.Y;
    }
}
